lib/event_store_client/connection.rb

```diff
@@ -29,17 +31,41 @@ def read(stream, direction: 'forward')
 
     def delete_stream(stream); end
 
+    def subscribe(stream, name:)
+      client.subscribe_to_stream(stream, name)
+    end
+
+    def consume_feed(stream, subscription)
+      response = client.consume_feed(stream, subscription)
+      return [] unless response.body
+      body = JSON.parse(response.body)
+      ack_uri = body['links'].find { |link| link['relation'] == 'ackAll' }.
+        try(:[], 'uri')
+      client.ack(ack_uri)
+      body['entries'].map do |entry|
+        event = EventStoreClient::Event.new(
+          id: entry['eventId'],
+          type: entry['eventType'],
+          data: entry['data'] || '{}',
+          metadata: entry['isMetaData'] ? entry['metaData'] : '{}'
```

Comment:
Because then you'll need to rely to pass hash. I really don't like that, as event initialization comes in two directions. One case is when you get a JSON data from event store, and second, when you get an event instance from the application.